src/api.cpp

```diff
@@ -401,38 +435,33 @@ compute_status faith_pd_one_off(const char* biom_filename, const char* tree_file
     return okay;
 }
 
+compute_status one_off_inmem(const support_biom_t *table_data, const support_bptree_t *tree_data,
+                             const char* unifrac_method, bool variance_adjust, double alpha,
+                             bool bypass_tips, unsigned int nthreads, mat_t** result) {
```

Comment:
You know the size of the resulting matrix at this point, so you would be better off to pass the pre-allocated buffer and fill it in, instead of letting one_off create one that you then need to make a copy of.
(even better if it was in plain-matrix, and not condensed format)